tests/cxx/tests/DREAM/MeanExcitationEnergy.cpp

```diff
@@ -0,0 +1,181 @@
+ */
+DREAM::IonHandler *MeanExcitationEnergy::GetIonHandler(
+    DREAM::FVM::Grid *g, DREAM::FVM::UnknownQuantityHandler *uqh, const len_t N_IONS, const len_t *Z_IONS
+) {
+    vector<string> tritiumNames(0);
+    vector<string> names(N_IONS);
+    for (len_t i = 0; i < N_IONS; i++)
+        names[i] = "";
+
+    return new DREAM::IonHandler(
+        g->GetRadialGrid(), uqh, Z_IONS, N_IONS, names, tritiumNames
+    );
+}
+
+void MeanExcitationEnergy::GetMeanExcitationEnergies(real_t *meanExcitationEnergies, DREAM::CollisionQuantity::collqty_settings *cq, const len_t N_IONS, const len_t *Z_IONS, 
+    const len_t N_SPECIES_TO_TEST, const len_t *Z_TO_TEST, const len_t *Z0_TO_TEST,
+    const real_t ION_DENSITY_REF, const real_t T_cold, const real_t B0, const len_t nr
+){
+    DREAM::FVM::Grid *grid = this->InitializeFluidGrid(nr,B0);
+    
+    DREAM::FVM::UnknownQuantityHandler *unknowns = GetUnknownHandler(grid,N_IONS, Z_IONS, ION_DENSITY_REF,T_cold);
+    DREAM::IonHandler *ionHandler = GetIonHandler(grid,unknowns, N_IONS, Z_IONS);
+    DREAM::OptionConstants::momentumgrid_type gridtype = DREAM::OptionConstants::MOMENTUMGRID_TYPE_PXI;
+
+    DREAM::CoulombLogarithm lnLEE(grid,unknowns,ionHandler,gridtype,cq,DREAM::CollisionQuantity::LNLAMBDATYPE_EE);
+    DREAM::CoulombLogarithm lnLEI(grid,unknowns,ionHandler,gridtype,cq,DREAM::CollisionQuantity::LNLAMBDATYPE_EI);
+    DREAM::SlowingDownFrequency nuS(grid,unknowns,ionHandler,&lnLEE,&lnLEI,gridtype,cq);
+    nuS.RebuildRadialTerms();
+    
+    len_t iz = 0;
+    for (len_t is = 0; is < N_SPECIES_TO_TEST; is++) {
+        if (Z_TO_TEST[is] != Z_IONS[iz]){ iz++; }
+        meanExcitationEnergies[is] = nuS.GetMeanExcitationEnergy(iz,Z0_TO_TEST[is]);
+    }
+
+    delete ionHandler;
+    delete unknowns;
+    delete grid;
+}
+
+bool MeanExcitationEnergy::CompareMeanExcitationEnergyWithTabulated(){
+
+    DREAM::CollisionQuantity::collqty_settings *cq =
+        new DREAM::CollisionQuantity::collqty_settings;
+
+    cq->collfreq_type = DREAM::OptionConstants::COLLQTY_COLLISION_FREQUENCY_TYPE_PARTIALLY_SCREENED;
+    cq->collfreq_mode = DREAM::OptionConstants::COLLQTY_COLLISION_FREQUENCY_MODE_SUPERTHERMAL;
+    cq->lnL_type      = DREAM::OptionConstants::COLLQTY_LNLAMBDA_ENERGY_DEPENDENT;
+    cq->bremsstrahlung_mode = DREAM::OptionConstants::EQTERM_BREMSSTRAHLUNG_MODE_STOPPING_POWER;
+
+    len_t nr = 1;
+
+    const len_t N_IONS = 3;
+    const len_t Z_IONS[N_IONS] = {10,18,36};
+    constexpr int_t N_SPECIES_TO_TEST = 10;
+    const len_t Z_TO_TEST[N_SPECIES_TO_TEST] =  {10,10,10,18,18,18,36,36,36,36};
+    const len_t Z0_TO_TEST[N_SPECIES_TO_TEST] = { 0, 1, 5, 0, 1, 9, 0, 1,18,30};
+
+    const real_t TABULATED_MEAN_EXCITATION_ENERGIES[] = {//Ne0, Ne1, Ne5, Ar0, Ar1, Ar9, Kr0, Kr1, Kr18, Kr30
+    2.68494e-04, 
+    3.23288e-04, 
+    6.90021e-04, 
+    3.69277e-04, 
+    4.29551e-04, 
+    1.55969e-03, 
+    7.04502e-04, 
+    8.06097e-04, 
+    3.45034e-03,
+    1.01486e-02
```

Comment:
I would put all of these on one line and place the comment on the previous just above it, and align the element names with the values.

Reply:
Do you mean like
const real_t TABULATED_MEAN_EXCITATION_ENERGIES[] = {
        //Ne0,            Ne1,         Ne5, ...       
        X,                  Y              Z, ...
    };

?

Reply:
Essentially. Exactly like this:
```
// Ne0,      Ne1          Ne5
2.68494e-04, 3.23288e-04, 6.90021e-04, ...
```